pkg/services/ngalert/metrics/state.go

```diff
@@ -0,0 +1,33 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type State struct {
+	GroupRules *prometheus.GaugeVec
+	AlertState *prometheus.GaugeVec
+}
+
+func newStateMetrics(r prometheus.Registerer) *State {
```

Comment:
Nit: We could export this, maybe? This seems like something we might want to consume from other places, we'll need it anywhere we want to construct a state manager.

Reply:
As discussed on Slack, ended up exporting all of them as the types are already public.